File: backend/src/main/java/com/magesty/backend/models/Experience.java
package com.magesty.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.magesty.backend.models.dto.ExperienceDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;
import java.util.Objects;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Experience implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String realisedAt;
    @Column
    private LocalDate start;
    @Column
    private LocalDate expected_end;
    private String description;

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn( name = "expId")
    private Admin admin;

    public static Experience from(ExperienceDto experienceDto){
        Experience experience = new Experience();

        if(Objects.isNull(experienceDto)){
            return null;
        }else{
            experience.setId(experienceDto.getId());
            experience.setDescription(experienceDto.getDescription());
            experience.setRealisedAt(experienceDto.getRealisedAt());
            experience.setStart(experienceDto.getStart());
            experience.setExpected_end(experienceDto.getExpected_end());

            return experience;
        }
    }
}

File: frontend-ng/src/app/models/Experience/experience.ts
export class Experience {
  private _id: number;
  private _realisedAt: string;
  private _start: Date;
  private _expected_end: Date;
  private _description: string;

  constructor(id: number, realisedAt: string, start: Date, expected_end: Date, description: string) {
    this._id = id;
    this._realisedAt = realisedAt;
    this._start = start;
    this._expected_end = expected_end;
    this._description = description;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get realisedAt(): string {
    return this._realisedAt;
  }

  set realisedAt(value: string) {
    this._realisedAt = value;
  }

  get start(): Date {
    return this._start;
  }

  set start(value: Date) {
    this._start = value;
  }

  get expected_end(): Date {
    return this._expected_end;
  }

  set expected_end(value: Date) {
    this._expected_end = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }
}
